phoenix-core/src/test/java/org/apache/phoenix/query/BaseTest.java

```diff
@@ -765,15 +767,18 @@ protected ResultSet executeQuery(Connection conn, QueryBuilder queryBuilder) thr
      * Note, we can't have this value too high since we don't want the shutdown to take too
      * long a time either.
      */
-    private static final int TEARDOWN_THRESHOLD = 30;
+    private static final int TEARDOWN_THRESHOLD = 1;
+
+    @ClassRule
+    public static PhoenixTestTableName phoenixTestTableName = new PhoenixTestTableName();
 
     public static String generateUniqueName() {
         int nextName = NAME_SUFFIX.incrementAndGet();
         if (nextName >= MAX_SUFFIX_VALUE) {
             throw new IllegalStateException("Used up all unique names");
         }
         TABLE_COUNTER.incrementAndGet();
-        return "N" + Integer.toString(MAX_SUFFIX_VALUE + nextName).substring(1);
+        return phoenixTestTableName.getTableName() + "_" + Integer.toString(MAX_SUFFIX_VALUE + nextName).substring(1);
```

Comment:
Some initial thoughts - 
1. Maybe we should use the new naming convention (tablename decorated) only for the distributed case. 
2. Do the regular ITs pass? when not qualified by a category?